nav2_system_tests/src/behaviors/assisted_teleop/assisted_teleop_behavior_tester.cpp

```diff
@@ -0,0 +1,280 @@
+    node_,
+    footprint_topic,
+    *tf_buffer_);
+
+  collision_checker_ = std::make_unique<nav2_costmap_2d::CostmapTopicCollisionChecker>(
+    *costmap_sub_,
+    *footprint_sub_
+  );
+
+  stamp_ = node_->now();
+}
+
+AssistedTeleopBehaviorTester::~AssistedTeleopBehaviorTester()
+{
+  if (is_active_) {
+    deactivate();
+  }
+}
+
+void AssistedTeleopBehaviorTester::activate()
+{
+  if (is_active_) {
+    throw std::runtime_error("Trying to activate while already active");
+    return;
+  }
+
+  while (!initial_pose_received_) {
+    RCLCPP_WARN(node_->get_logger(), "Initial pose not received");
+    sendInitialPose();
+    std::this_thread::sleep_for(100ms);
+    rclcpp::spin_some(node_);
+  }
+
+  // Wait for lifecycle_manager_navigation to activate behavior_server
+  std::this_thread::sleep_for(10s);
+
+  if (!client_ptr_) {
+    RCLCPP_ERROR(node_->get_logger(), "Action client not initialized");
+    is_active_ = false;
+    return;
+  }
+
+  if (!client_ptr_->wait_for_action_server(10s)) {
+    RCLCPP_ERROR(node_->get_logger(), "Action server not available after waiting");
+    is_active_ = false;
+    return;
+  }
+
+  RCLCPP_INFO(this->node_->get_logger(), "Assisted Teleop action server is ready");
+  is_active_ = true;
+}
+
+void AssistedTeleopBehaviorTester::deactivate()
+{
+  if (!is_active_) {
+    throw std::runtime_error("Trying to deactivate while already inactive");
+  }
+  is_active_ = false;
+}
+
+bool AssistedTeleopBehaviorTester::defaultAssistedTeleopTest(
+  const float lin_vel,
+  const float ang_vel)
+{
+  if (!is_active_) {
+    RCLCPP_ERROR(node_->get_logger(), "Not activated");
+    return false;
+  }
+
+  // Sleep to let behavior server be ready for serving in multiple runs
+  std::this_thread::sleep_for(1s);
+
+  RCLCPP_INFO(node_->get_logger(), "Sending goal");
+
+  auto goal_handle_future = client_ptr_->async_send_goal(nav2_msgs::action::AssistedTeleop::Goal());
+
+  if (rclcpp::spin_until_future_complete(node_, goal_handle_future) !=
+    rclcpp::FutureReturnCode::SUCCESS)
+  {
+    RCLCPP_ERROR(node_->get_logger(), "send goal call failed :(");
+    return false;
+  }
+
+  rclcpp_action::ClientGoalHandle<AssistedTeleop>::SharedPtr goal_handle = goal_handle_future.get();
+  if (!goal_handle) {
+    RCLCPP_ERROR(node_->get_logger(), "Goal was rejected by server");
+    return false;
+  }
+
+  // Wait for the server to be done with the goal
+  auto result_future = client_ptr_->async_get_result(goal_handle);
+
+  rclcpp::Rate r(1);
+
+  counter_ = 0;
+  auto start_time = std::chrono::system_clock::now();
+  while (rclcpp::ok()) {
+    geometry_msgs::msg::Twist cmd_vel = geometry_msgs::msg::Twist();
+    cmd_vel.linear.x = lin_vel;
+    cmd_vel.angular.z = ang_vel;
+    cmd_vel_pub_->publish(cmd_vel);
+
+    if (counter_ > 1) {
+      break;
+    }
+
+    auto current_time = std::chrono::system_clock::now();
+    if (current_time - start_time > 25s) {
```

Comment:
Is that how long it takesish?

Reply:
I can reduce this. We currently have three tests that take place. 

1) collide backward, just lin_vel
2) collide forward, neg lin vel
3) collide while driving will ang and lin vel.

The collide forward takes a while because it drives across the world.